### PR TITLE
Spotaut 9182 sdk saikumar

### PR DESCRIPTION
**Pull request by Saikumar-Jalda**
_This is the default pull request template. You can customize it by adding a `pull_request_template.md` at the root of your repo or inside the `.github` folder._

# Jira Ticket

_Include a link to your Jira Ticket_
_Example: [JIRAISS-1234](https://spotinst.atlassian.net/browse/JIRAISS-1234)_

# Demo

_Please add a recording of the feature/bug fix in work. if you added new routes, the recording should show the request and response for each new/changed route_

# Checklist:
- [ ] I have filled relevant self assessment ([NodeJS](https://docs.google.com/forms/d/e/1FAIpQLSfl14u9AOBAmxVJ272tvO7XNuXE-EMvWaGcaRZalu1UAKB7RA/viewform), [Frontend](https://docs.google.com/forms/d/e/1FAIpQLSdiBPNKH81w_EkavihVL8Uwb0j7tP8PwJmLFYm2nCOQxz-1qw/viewform), [Backend](https://docs.google.com/forms/d/e/1FAIpQLSed_PsTJ5-XIWkFL6BSDE2AQRBVPmwc3PAmHZkUn-erzVI37Q/viewform?usp=sf_link))
- [ ] I have run ESlint on my changes and fixed all warnings and errors (**NodeJS & Frontend Services**)
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have validated all the requirements in the Jira task were answered
- [ ] I have all neccessary approvals for the design/mini design of this task
- [ ] I have approved the API changes and granular permission patterns (documentation subtask) (**For public services only**)

### CODE REVIEW

File: src/main/java/com/spotinst/sdkjava/model/ISpotinstElastigroupRepoGcp.java [hunk 1]
```diff
@@ -1,6 +1,20 @@
...
 import com.spotinst.sdkjava.model.bl.gcp.ElastigroupGcp;
+import com.spotinst.sdkjava.model.requests.elastigroup.gcp.ElastigroupInstanceLockRequestGcp;
+import com.spotinst.sdkjava.model.requests.elastigroup.gcp.ElastigroupInstanceUnLockRequestGcp;
+import com.spotinst.sdkjava.model.requests.elastigroup.gcp.ElastigroupScalingRequestGcp;
+import com.spotinst.sdkjava.model.responses.elastigroup.gcp.ElastigroupScalingResponseGcp;
 
 interface ISpotinstElastigroupRepoGcp extends IRepositoryGcp<ElastigroupGcp, GroupFilter, String> {
+    RepoGenericResponse<Boolean> lockInstance(ElastigroupInstanceLockRequestGcp lockRequest,
+                                              String authToken, String instanceId);
+
```
Comment (janetlinkiruba) [inline src/main/java/com/spotinst/sdkjava/model/ISpotinstElastigroupRepoGcp.java:12]: why do we need to prepare request class for lock/unlock /scaleup/scale down?

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClientGcp.java [hunk 2]
```diff
@@ -191,5 +206,112 @@ public ElastigroupGcp getElastigroup(ElastigroupGetRequestGcp elastigroupGetRequ
...
+    }
+
+    /*
+     * This function is used to scale up an Elastigroup by taking an ElastigroupScalingRequest that is sent by the user and sending it to
+     * ISpotinstElastigroupRepoGcp.scaleUp().
+     *
+     * @param elastigroupScalingRequestGcp ElastigroupScalingRequestGcp object that can be converted to a JSON to send as a request
+     * @return ElastigroupScalingResponseGcp Object that is returned from the scale down request from ISpotinstElastigroupRepo.scaleUp()
+     */
+    public ElastigroupScalingResponseGcp scaleGroupUp(ElastigroupScalingRequestGcp elastigroupScalingRequestGcp) {
```
Comment (janetlinkiruba) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClientGcp.java:260]: Rename the function names to scaleUpGroup and scaleDownGroup

File: src/main/java/com/spotinst/sdkjava/example/elastigroup/gcp/ElastigroupUsageExampleGcp.java [hunk 3]
```diff
@@ -41,8 +41,7 @@ public static void main(String[] args) throws IOException {
         System.out.println(String.format(preFormat, elastigroupId, groupName));
 
         // Sleep for provisioning
-        System.out.println("\nSleeping... waiting for provisioning 7 seconds.");
-        sleep(7);
+        System.out.println("\nSleeping... waiting for provisioning 7 seconds.");sleep(7);
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/example/elastigroup/gcp/ElastigroupUsageExampleGcp.java:44]: sleep statement should be in next line.

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupCreationRequestGcp.java [hunk 4]
```diff
@@ -13,6 +13,7 @@
 public class ElastigroupCreationRequestGcp {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupCreationRequestGcp.java:13]: This should be moved to requests folder

File: src/main/java/com/spotinst/sdkjava/model/ElastigroupGetAllRequestGcp.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
+package com.spotinst.sdkjava.model;
+
+import com.spotinst.sdkjava.exception.SpotinstValidationException;
+
+import java.util.Date;
+
+public class ElastigroupGetAllRequestGcp {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/ElastigroupGetAllRequestGcp.java:7]: This class is not needed.

File: src/main/java/com/spotinst/sdkjava/model/IRepositoryGcp.java [hunk 6]
```diff
@@ -14,6 +14,10 @@
         throw new SpotinstNotSupportedException();
     }
 
+    default RepoGenericResponse<List<S>> getAll(F filter, String authToken, String account) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/IRepositoryGcp.java:17]: This is not needed.

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClientGcp.java [hunk 7]
```diff
@@ -144,6 +158,34 @@ public Boolean deleteElastigroup(ElastigroupDeletionRequestGcp elastigroupDeleti
         return retVal;
     }
 
+    public List<ElastigroupGcp> getAllElastigroups(ElastigroupGetAllRequestGcp elastigroupGetAllRequest) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClientGcp.java:161]: This is not needed.

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClientGcp.java [hunk 8]
```diff
@@ -17,8 +20,17 @@
     //region Members
     private              String                                    authToken;
     private              String                                    account;
+    private              ISpotinstElastigroupRepoGcp               spotinstElastigroupRepoGcp;
     //endregion
 
+    public ISpotinstElastigroupRepoGcp getSpotinstElastigroupRepoGcp() {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupClientGcp.java:26]: If we are adding this common constructor to fetch repo object, make the change in all APIs in this class to use getSpotinstElastigroupRepoGcp() function to fetch repo object.

Reference - similar class for Azure.

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupServiceGcp.java [hunk 9]
```diff
@@ -132,6 +135,68 @@ public static Boolean deleteElastigroup(String elastigroupId, String authToken,
         return retVal;
     }
 
+    public static List<ApiElastigroupGcp> getAllGroups(GroupFilter filter, String authToken,
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupServiceGcp.java:138]: To be removed.

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupServiceGcp.java [hunk 10]
```diff
@@ -242,4 +307,158 @@ public static ApiElastigroupGcp getGroup(String elastigroupId, String authToken,
...
+        Map<String, String> headers = buildHeaders(authToken);
+
+        // Build URI
+        String uri = String.format("%s/gcp/gce/instance/%s/unlock", apiEndpoint, instanceId);
+
+        // Send the request.
+        RestResponse response = RestClient.sendPost(uri, null, headers, queryParams);
+
+        // Handle the response.
+        ElastigroupApiResponseGcp unlockResponse = getCastedResponse(response, ElastigroupApiResponseGcp.class);
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupServiceGcp.java:368]: Reponse type should be BaseServiceEmptyResponse

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupServiceGcp.java [hunk 11]
```diff
@@ -242,4 +307,158 @@ public static ApiElastigroupGcp getGroup(String elastigroupId, String authToken,
...
+        Map<String, String> headers = buildHeaders(authToken);
+
+        // Build URI
+        String uri = String.format("%s/gcp/gce/instance/%s/lock", apiEndpoint, instanceId);
+
+        // Send the request.
+        RestResponse response = RestClient.sendPost(uri, null, headers, queryParams);
+
+        // Handle the response.
+        ElastigroupApiResponseGcp lockResponse = getCastedResponse(response, ElastigroupApiResponseGcp.class);
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupServiceGcp.java:336]: Reponse type should be BaseServiceEmptyResponse

File: src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupServiceGcp.java [hunk 12]
```diff
@@ -242,4 +307,158 @@ public static ApiElastigroupGcp getGroup(String elastigroupId, String authToken,
...
+        // Handle the response.
+        ElastigroupApiResponseGcp unlockResponse = getCastedResponse(response, ElastigroupApiResponseGcp.class);
+        if (unlockResponse.getResponse().getStatus().getCode() == HttpStatus.SC_OK) {
+            retVal = true;
+        }
+
+        return retVal;
+    }
+
+    /*
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupServiceGcp.java:376]: please remove comments.

File: src/main/java/com/spotinst/sdkjava/model/api/gcp/ApiElastigroupScaleDownResponseGcp.java [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiElastigroupScaleDownResponseGcp implements IPartialUpdateEntity {
+
+    @JsonIgnore
+    private Set<String>                                     isSet;
+    private List<ApiElastigroupScaleDownVictimPreemptiblesListGcp>  victimPreemptiblesList;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/gcp/ApiElastigroupScaleDownResponseGcp.java:20]: Change class name from ApiElastigroupScaleDownVictimPreemptiblesListGcp to ApiElastigroupScaleDownVictimPreemptiblesGcp

File: src/main/java/com/spotinst/sdkjava/model/api/gcp/ApiElastigroupScaleDownResponseGcp.java [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+
+    public Set<String> getIsSet() {
+        return isSet;
+    }
+
+    public void setIsSet(Set<String> isSet) {
+        this.isSet = isSet;
+    }
+
+    public List<ApiElastigroupScaleDownVictimPreemptiblesListGcp> getVictimPreemptibles() {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/gcp/ApiElastigroupScaleDownResponseGcp.java:35]: Rename to getVictimPreemptiblesList()

File: src/main/java/com/spotinst/sdkjava/model/api/gcp/ApiElastigroupScaleDownVictimInstancesListGcp.java [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ApiElastigroupScaleDownVictimInstancesListGcp {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/gcp/ApiElastigroupScaleDownVictimInstancesListGcp.java:12]: Change class name to ApiElastigroupScaleDownVictimInstancesGcp

File: src/main/java/com/spotinst/sdkjava/model/api/gcp/ApiElastigroupScaleDownVictimPreemptiblesListGcp.java [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ApiElastigroupScaleDownVictimPreemptiblesListGcp {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/gcp/ApiElastigroupScaleDownVictimPreemptiblesListGcp.java:12]: Same comment as above.

File: src/main/java/com/spotinst/sdkjava/model/api/gcp/ApiElastigroupScaleUpNewInstancesListGcp.java [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ApiElastigroupScaleUpNewInstancesListGcp {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/gcp/ApiElastigroupScaleUpNewInstancesListGcp.java:12]: Rename to ApiElastigroupScaleUpNewInstancesGcp

File: src/main/java/com/spotinst/sdkjava/model/api/gcp/ApiElastigroupScaleUpNewPreemptiblesListGcp.java [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ApiElastigroupScaleUpNewPreemptiblesListGcp {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/gcp/ApiElastigroupScaleUpNewPreemptiblesListGcp.java:12]: Redundant class. Remove this and use the above class.

File: src/main/java/com/spotinst/sdkjava/model/api/gcp/ApiElastigroupScaleUpResponseGcp.java [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiElastigroupScaleUpResponseGcp implements IPartialUpdateEntity {
+
+    @JsonIgnore
+    private Set<String>                                     isSet;
+    private List<ApiElastigroupScaleUpNewPreemptiblesListGcp>  newPreemptiblesList;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/gcp/ApiElastigroupScaleUpResponseGcp.java:20]: The class names use for creating list object will change as per above comments.

File: src/main/java/com/spotinst/sdkjava/model/bl/gcp/ElastigroupCapacityConfigurationGcp.java [hunk 20]
```diff
@@ -0,0 +1,120 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ElastigroupCapacityConfigurationGcp {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/bl/gcp/ElastigroupCapacityConfigurationGcp.java:13]: Remove this class.

File: src/main/java/com/spotinst/sdkjava/model/bl/gcp/ElastigroupCapacityGcp.java [hunk 21]
```diff
@@ -8,14 +8,15 @@
 import java.util.HashSet;
 import java.util.Set;
 
+@JsonIgnoreProperties(ignoreUnknown = true)
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/bl/gcp/ElastigroupCapacityGcp.java:11]: There shouldn't be no change in this class.

File: src/main/java/com/spotinst/sdkjava/model/bl/gcp/ElastigroupScaleDownResponseGcp.java [hunk 22]
```diff
@@ -0,0 +1,88 @@
...
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/bl/gcp/ElastigroupScaleDownResponseGcp.java:11]: All BL class comments should be same as Api classes.

File: src/main/java/com/spotinst/sdkjava/model/converters/elastigroup/gcp/ElastigroupConverterGcp.java [hunk 23]
```diff
@@ -282,32 +283,24 @@ private static ApiRevertToPreemptibleGcp toDal(ElastigroupRevertToPreemptibleGcp
         return retVal;
     }
 
-    private static ApiCapacityGcp toDal(ElastigroupCapacityGcp capacityConfigurationGcp) {
+    private static ApiCapacityGcp toDal(ElastigroupCapacityGcp capacity) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/converters/elastigroup/gcp/ElastigroupConverterGcp.java:286]: This function shouldn't be changed.

File: src/main/java/com/spotinst/sdkjava/model/converters/elastigroup/gcp/ElastigroupConverterGcp.java [hunk 24]
```diff
@@ -693,5 +683,119 @@ private static ElastigroupInstanceTypesGcp toBl(ApiInstanceTypesGcp instanceType
...
         return retVal;
     }
+
+    public static ElastigroupScaleUpResponseGcp toBl(ApiElastigroupScaleUpResponseGcp elastigroupScaleUpResponseGcp) {
+        ElastigroupScaleUpResponseGcp retVal = null;
+
+        if (elastigroupScaleUpResponseGcp != null) {
+            retVal = new ElastigroupScaleUpResponseGcp();
+
+            if (elastigroupScaleUpResponseGcp.isNewPreemptiblesSet()) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/converters/elastigroup/gcp/ElastigroupConverterGcp.java:693]: Class names will change as per other comments.

File: src/main/java/com/spotinst/sdkjava/model/converters/elastigroup/gcp/ElastigroupConverterGcp.java [hunk 25]
```diff
@@ -693,5 +683,119 @@ private static ElastigroupInstanceTypesGcp toBl(ApiInstanceTypesGcp instanceType
...
+            if (elastigroupScaleUpNewPreemptiblesListGcp.isInstanceNameSet()) {
+                retVal.setInstanceName(elastigroupScaleUpNewPreemptiblesListGcp.getInstanceName());
+            }
+
+        }
+        return retVal;
+
+    }
+
+    private static ElastigroupScaleUpNewInstancesListGcp toBl(
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/converters/elastigroup/gcp/ElastigroupConverterGcp.java:727]: This is redundant, if other comments are implemented.

File: src/main/java/com/spotinst/sdkjava/model/converters/elastigroup/gcp/ElastigroupConverterGcp.java [hunk 26]
```diff
@@ -693,5 +683,119 @@ private static ElastigroupInstanceTypesGcp toBl(ApiInstanceTypesGcp instanceType
...
+
+            if (elastigroupScaleUpNewInstancesListGcp.isInstanceNameSet()) {
+                retVal.setInstanceName(elastigroupScaleUpNewInstancesListGcp.getInstanceName());
+            }
+        }
+        return retVal;
+
+    }
+
+    public static ElastigroupScaleDownResponseGcp toBl(ApiElastigroupScaleDownResponseGcp elastigroupScaleDownResponseGcp) {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/converters/elastigroup/gcp/ElastigroupConverterGcp.java:743]: Class names will change.

File: src/main/java/com/spotinst/sdkjava/model/converters/elastigroup/gcp/ElastigroupConverterGcp.java [hunk 27]
```diff
@@ -693,5 +683,119 @@ private static ElastigroupInstanceTypesGcp toBl(ApiInstanceTypesGcp instanceType
...
+
+        }
+        return retVal;
+
+    }
+
+    private static ElastigroupScaleDownVictimInstancesListGcp toBl(
+            ApiElastigroupScaleDownVictimInstancesListGcp elastigroupScaleDownVictimInstancesListGcp) {
+        ElastigroupScaleDownVictimInstancesListGcp retVal = null;
+
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/converters/elastigroup/gcp/ElastigroupConverterGcp.java:787]: Redundant function.

File: src/main/java/com/spotinst/sdkjava/model/responses/elastigroup/gcp/ElastigroupUpdateCapacityApiResponseGcp.java [hunk 28]
```diff
@@ -0,0 +1,8 @@
+package com.spotinst.sdkjava.model.responses.elastigroup.gcp;
+
+import com.spotinst.sdkjava.client.response.BaseServiceItemsResponse;
+import com.spotinst.sdkjava.model.ApiElastigroupUpdateCapacityResponse;
+
+public class ElastigroupUpdateCapacityApiResponseGcp extends BaseServiceItemsResponse<ApiElastigroupUpdateCapacityResponse> {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/responses/elastigroup/gcp/ElastigroupUpdateCapacityApiResponseGcp.java:6]: This class is not required. Please remove.

File: src/main/java/com/spotinst/sdkjava/model/bl/gcp/ElastigroupCapacityConfigurationGcp.java (hunk 20 above)
Comment (Saikumar-Jalda) [reply src/main/java/com/spotinst/sdkjava/model/bl/gcp/ElastigroupCapacityConfigurationGcp.java:13]: already removed, not sure why we are seeing here.

File: src/main/java/com/spotinst/sdkjava/model/bl/gcp/ElastigroupScaleDownResponseGcp.java (hunk 22 above)
Comment (Saikumar-Jalda) [reply src/main/java/com/spotinst/sdkjava/model/bl/gcp/ElastigroupScaleDownResponseGcp.java:11]: is it removing of redundant class similar as api?  If  yes that is done.

File: src/main/java/com/spotinst/sdkjava/model/api/gcp/ApiElastigroupScaleDownResponseGcp.java [hunk 29]
```diff
@@ -0,0 +1,63 @@
...
+
+    public Set<String> getIsSet() {
+        return isSet;
+    }
+
+    public void setIsSet(Set<String> isSet) {
+        this.isSet = isSet;
+    }
+
+    public List<ApiElastigroupScaleDownVictimInstancesPreemptiblesGcp> getVictimPreemptiblesList() {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/gcp/ApiElastigroupScaleDownResponseGcp.java:35]: The function names should be changed to getVictimPreemptibles()

File: src/main/java/com/spotinst/sdkjava/model/api/gcp/ApiElastigroupScaleUpNewInstancesPreemptiblesGcp.java [hunk 30]
```diff
@@ -0,0 +1,70 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ApiElastigroupScaleUpNewInstancesPreemptiblesGcp {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/gcp/ApiElastigroupScaleUpNewInstancesPreemptiblesGcp.java:12]: Can be renamed to ApiElastigroupScaleUpNewInstancesGcp

File: src/main/java/com/spotinst/sdkjava/model/api/gcp/ApiElastigroupScaleUpResponseGcp.java [hunk 31]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonFilter("PartialUpdateEntityFilter")
+public class ApiElastigroupScaleUpResponseGcp implements IPartialUpdateEntity {
+
+    @JsonIgnore
+    private Set<String>                                     isSet;
+    private List<ApiElastigroupScaleUpNewInstancesPreemptiblesGcp>  newPreemptiblesList;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/api/gcp/ApiElastigroupScaleUpResponseGcp.java:20]: Keep variable names same as in response JSON.

File: src/main/java/com/spotinst/sdkjava/model/bl/elastigroup/gcp/ElastigroupScaleDownResponseGcp.java [hunk 32]
```diff
@@ -0,0 +1,88 @@
...
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ElastigroupScaleDownResponseGcp {
+
+    @JsonIgnore
+    private Set<String>                                     isSet;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/bl/elastigroup/gcp/ElastigroupScaleDownResponseGcp.java:16]: Same comments as API classes.

File: src/main/java/com/spotinst/sdkjava/model/bl/elastigroup/gcp/ElastigroupScaleUpResponseGcp.java [hunk 33]
```diff
@@ -0,0 +1,88 @@
...
+import java.util.List;
+import java.util.Set;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ElastigroupScaleUpResponseGcp {
+
+    @JsonIgnore
+    private Set<String>                                     isSet;
+    private List<ElastigroupScaleUpNewInstancesPreemptiblesGcp>  newPreemptiblesList;
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/bl/elastigroup/gcp/ElastigroupScaleUpResponseGcp.java:17]: Rename the private members.

**Review by anuragsharma-123 — APPROVED** [review]

**Pull request merged**